types/node/index.d.ts

```diff
@@ -477,6 +477,14 @@ declare namespace NodeJS {
         getuid(): number;
         setuid(id: number): void;
         setuid(id: string): void;
+        geteuid(): number;
+        seteuid(id: number): void;
+        seteuid(id: string): void;
```

Comment:
Please use an union type instead an overload, e.a. `seteuid(id: string | number): void`.
Would be nice if you could adapt the nearby functions `setgid` and `setuid` also...

Reply:
Why is a union type better than an overload?

Reply:
It's the way how typescript does the lookup and type detection. If you use an union type it's one function with one signature, if you use overloads then they are two functions.
Following does not transpile currently but works if union types are used:
```
function foo<T>(f: (p: T) => void, p: T) {
    return f(p);
}
foo(process.setgid, "foo"); // OK
foo(process.setgid, 5); // Argument of type '5' is not assignable to parameter of type 'string'
```
Typescript select always the last overload in such situations.